sixtools/info_functions.py

```diff
@@ -0,0 +1,86 @@
+from blueksy import simulators
+
+
+def list_scans(plan):
+    '''Print a summary of a plan, with some SIX specific info
+
+    Prints a version of the plan, showing moves that occur outside of runs, and
+    potential scan id's for any runs. Note that the scan id's listed assume
+    that the first scan has the next scan_id inthe sequence, if there is an
+    aborted scan, or if other scans occur first, this may not be true.
+
+    If a 'reason' kwarg is included in the metadata this is also included in
+    the printout.
+
+    For example the expected printout for each run should be:
+    "scan no XXXX, 'reason', motor1 = Y1, motor2 = Y2, ......."
+        where XXXX is the expected scan ID, 'reason' is an optional string
+        given by the 'reason' kwarg in he scans metadata and Yi is the value
+        that motor1 was set too prior to running the scan.
+
+    Parameters
+    ----------
+    plan: iterable
+        must yield'Msg' objects
+
+    '''
+
+    in_run = 0  # A way to track if we are in a run or not.
+    prev_scanID = db[-1].start['scan_id']  # The previous scan ID
```

Comment:
I am not a fan of this at all.  Can we format this as (literally) 'XXX' or 'N + i' ?

Reply:
@tacaswell this is the major driver for SIX, they want a 'summarize plans' that lists the expected scan id's. I have documented and mentioned to them that there is absolutely no guarantees this will be correct but they still wanted this included.